Core/include/Acts/EventData/MultiTrajectoryHelpers.hpp

```diff
@@ -44,21 +45,27 @@ using VolumeTrajectoryStateContainer =
 /// @brief Getter for global trajectory info
 ///
 /// @param multiTraj The MultiTrajectory object
-/// @param entryIndex The entry index of trajectory to investigate
+/// @param tipIndex The entry index of trajectory to investigate
 ///
 /// @return The trajectory summary info
 template <typename traj_t>
-TrajectoryState trajectoryState(const traj_t& multiTraj,
-                                std::size_t entryIndex) {
+TrajectoryState trajectoryState(const traj_t& multiTraj, std::size_t tipIndex) {
   TrajectoryState trajState;
-  multiTraj.visitBackwards(entryIndex, [&](const auto& state) {
+  multiTraj.visitBackwards(tipIndex, [&](const auto& state) {
     // Get the volume Id of this surface
     const auto& geoID = state.referenceSurface().geometryId();
     const auto& volume = geoID.volume();
     const auto& layer = geoID.layer();
     trajState.nStates++;
     auto typeFlags = state.typeFlags();
-    if (typeFlags.test(Acts::TrackStateFlag::MeasurementFlag)) {
```

Comment:
is this reordering of the `if` clauses (here and elsewhere) because we can have a measurement and an outlier (ie. both bits set) for the same track state. Is that expected? I don't think I saw that before, when I printed out the flags.

Reply:
I think that was the intention of having flags there yes. But that freedom can also be a problem as we discover here in the code and from just thinking differently about it.

Generally the rotation of the `if` should provide a more stable solution